lighthouse-core/lib/network-recorder.js

```diff
@@ -304,15 +304,22 @@ class NetworkRecorder extends EventEmitter {
     request.onResourceChangedPriority(data);
   }
 
+  /**
+   * @param {LH.Crdp.Target.ReceivedMessageFromTargetEvent} data
+   */
+  onReceivedMessageFromTarget(data) {
+    /** @type {LH.Protocol.RawMessage} */
+    const protocolMessage = JSON.parse(data.message);
+    if ('id' in protocolMessage) return;
+
+    this.dispatch(protocolMessage);
+  }
+
   /**
    * Routes network events to their handlers, so we can construct networkRecords
    * @param {LH.Protocol.RawEventMessage} event
    */
   dispatch(event) {
-    if (!event.method.startsWith('Network.')) {
```

Comment:
in case you're curious like me if this was saving us anytime, it looks like it wasn't

just a switch is ~10x faster

https://jsperf.com/switch-vs-startswith